### PR TITLE
Milestone 3. Game logic.

### PR DESCRIPTION
**Pull request by Luzaks**
We build the logic for the tic-tac-toe-game.

### CODE REVIEW

File: lib/logic.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    when 1..3
+      @@board[0][@number - 1] = @@token
+      @array_of_number << @number
+    when 4..6
+      puts @@board[1][@number - 4] = @@token
+      @array_of_number << @number
+    when 7..9
+      puts @@board[2][@number - 7] = @@token
+      @array_of_number << @number
+    end
```
Comment (stratospherique) [inline lib/logic.rb]: image
* [x] No puts/print and get.chomp outside main.rb file.

File: lib/logic.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+class GameLogic #:nodoc:
+  attr_accessor :number, :board, :turn, :array_of_number
```
Comment (stratospherique) [inline lib/logic.rb]: * [x] Use attr_readers/attr_writers instead of attr_accessors.

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great start @Team :clap: :clap: However, I have some suggestion that you need to implement in order to get approval for this milestone. check :arrow_down: 
##### OOP & Project design

* [ ] The project uses at least two classes (Usually it uses three: Game, Board and Player). The code of the project should be logically divided between classes. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/classes.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
* [ ] All classes are kept in `lib` directory. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

  - You should think of decoupling the game into concepts that can be translated into classes (objects) able to cummunicate between each other via messages (methods).

- **[REQUIRED]** check code comments below, use the notes to refactor your code :pray: 

### That's it :relaxed: When finished, kindly resubmit another code review request :pray: 

Happy coding :v: 
_**Ahmed Mahfoudh**_

File: lib/logic.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+class GameLogic #:nodoc:
+  attr_writer :number, :turn, :array_of_number, :board
```
Comment (ebukaume) [inline lib/logic.rb]: Why would you want to write these attribute from the outside? For example; one can overwrite `turn` without going through `#change_turn`

File: bin/main.rb [hunk 4]
```diff
@@ -1,71 +1,84 @@
...
   def initialize
-    puts 'You are about to start the Tic Tac Toe game.'
-    puts "What's the name of the first player?"
+    puts ' You are about to start the Tic Tac Toe game.'
+    puts " What's the name of the first player?"
     @player1 = gets.chomp
-    puts "What's the name of the second player?"
+    puts " What's the name of the second player?"
     @player2 = gets.chomp
-
```
Comment (ebukaume) [inline bin/main.rb]: Move this class away from `bin/` to `lib/`

**Review by ebukaume — CHANGES_REQUESTED** [review]
## Changes Required!

Hi team :wave: 

Nice of you making some of the changes, but there is more. All your classes should be in the `lib/` directory. Also, see below for more.

**Kindly submit another code review after making the fix**

Happy Coding :v: 
**Ebuka Umeokonkwo**
 [Github](https://github.com/ebukaume) | [Linkedin](https://www.linkedin.com/in/ebukaume) | [Slack](https://app.slack.com/team/UFDKXE68L) | [Twitter](https://twitter.com/ebukaume)

**Review by ebukaume — CHANGES_REQUESTED** [review]
## Changes Required! :stop_sign: 

Hi team :wave: 

I see you have removed the `attr_writer`. However, you guys still need to move **all** class away from `bin/` and into `lib/`. 

- [x] Also, please you the [linter configuration](https://github.com/microverseinc/linters-config/tree/master/ruby) provided by Microverse. Please remove all other linter config files.
- [x] Also, fix the permission issue on `bin/main.rb` file.
review-permission
- [x] Finally, don't forget to remove the `.idea` directory and move the `.gitignore` to the root of the repo.

_I am sorry we didn't mention the last 3 in the previous reviews_

**Kindly submit another code review request after making those changes**. You can also reach me on slack if you need further assistance.

Happy Coding :v: 
**Ebuka Umeokonkwo**
 [Github](https://github.com/ebukaume) | [Linkedin](https://www.linkedin.com/in/ebukaume) | [Slack](https://app.slack.com/team/UFDKXE68L) | [Twitter](https://twitter.com/ebukaume)

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi team!

### Amazing job thus far! :raised_hands: I see you have fixed all the other TSE requests.

However, the linter is not properly set up. You don't need to have multiple `.rubocop.yml` or `.rubocop_todo.yml` files. Kindly fix this as appropriate. See [this guide](https://github.com/microverseinc/linters-config/tree/master/ruby) to set it up.

**Please submit for another code review once this has been fixed.** _Happy coding!_ :tada:

**Review by ebukaume — APPROVED** [review]
## Approved! :champagne: 

You gals are really doing great. Please keep the good work up. **You can go ahead and merge this branch.**

Happy Coding :v: 
**Ebuka Umeokonkwo**
 [Github](https://github.com/ebukaume) | [Linkedin](https://www.linkedin.com/in/ebukaume) | [Slack](https://app.slack.com/team/UFDKXE68L) | [Twitter](https://twitter.com/ebukaume)

**Pull request merged**